src/configurationProvider.ts

```diff
@@ -34,7 +34,7 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
         return vscode.window.withProgress({location: vscode.ProgressLocation.Window}, (p) => {
             return new Promise((resolve, reject) => {
                 p.report({message: "Auto generating configuration..."});
-                resolveMainClass(folder.uri).then((res: any[]) => {

```

Comment:
In which case the folder in empty? Single file? 

Does the java-debug also should be changed for this changes?

Reply:
if client only opens a file, the folder is `undefined`. java-debug has handled this.